engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/runner/FlinkScenarioJob.scala

```diff
@@ -40,7 +43,26 @@ class FlinkScenarioJob(modelData: ModelData) {
       env: StreamExecutionEnvironment,
       processListeners: List[ProcessListener],
   ): JobExecutionResult = {
-    val compilerFactory         = new FlinkProcessCompilerDataFactory(modelData, deploymentData, processListeners)
+    val liveDataCollectingListener =
+      modelData.modelConfig.liveDataPreviewMode match {
+        case LiveDataPreviewMode.Disabled =>
+          None
+        case LiveDataPreviewMode.Enabled(maxNumberOfSamples, throughputTimeWindowInSeconds, dbUploaderOpt) =>
+          val processIdWithName = ProcessIdWithName(processVersion.processId, processVersion.processName)
+          dbUploaderOpt.foreach(PeriodicLiveDataUploader.register(env, processIdWithName, _))
+          Some(
+            LiveDataCollectingListenerHolder.createListenerFor(
```

Comment:
When do you close the allocated resources for jobs that have been run on a remote flink cluster?